tests/unit/Services/ApmCollectorServiceTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+use AG\ElasticApmLaravel\Agent;
+use AG\ElasticApmLaravel\Contracts\DataCollector;
+use AG\ElasticApmLaravel\Services\ApmCollectorService;
+use Codeception\Test\Unit;
+use Illuminate\Config\Repository as Config;
+use Illuminate\Events\Dispatcher;
+use Illuminate\Foundation\Application;
+
+class ApmCollectorServiceTest extends Unit
+{
+    private const SPAN_NAME = 'test-measure';
+
+    private $appMock;
+    private $configMock;
+    private $eventsMock;
+
+    private $collectorService;
+
+    protected function setUp(): void
+    {
+        parent::setup();
+
+        $this->appMock = Mockery::mock(Application::class);
+        $this->configMock = Mockery::mock(Config::class);
+        $this->eventsMock = Mockery::mock(Dispatcher::class);
+
+        $this->configMock->shouldReceive('get')
+            ->once()
+            ->with('elastic-apm-laravel.active')
+            ->andReturn(true);
+
+        $this->collectorService = new ApmCollectorService(
+            $this->appMock,
+            $this->eventsMock,
+            $this->configMock
+        );
+    }
+
+    public function testStartMeasureEventWithDefaultValues()
+    {
+        $this->eventsMock->shouldReceive('dispatch')
+            ->once()
+            ->with(Mockery::on(function ($event) {
+                $this->assertEquals(self::SPAN_NAME, $event->name);
+                $this->assertEquals('request', $event->type);
+                $this->assertNull($event->action);
+                $this->assertNull($event->label);
+                $this->assertNull($event->start_time);
+
+                return true;
+            }));
+
+        $this->collectorService->startMeasure(self::SPAN_NAME);
+    }
+
+    public function testStartMeasureEventWithValues()
+    {
+        $this->eventsMock->shouldReceive('dispatch')
```

Comment:
This accomplishes the same thing but another good way to assert events are dispatched is to register a listener inside the test method and set a value to check after the SUT runs.

Reply:
I'm not sure what you mean. Do you have a concrete example?

Reply:
I’m traveling and on my phone so this is real rough...

$dispatched = false;
App()->listen(MyEvent::class, function() {
  $dispatched = true;
});

$collector->startMeasure(...);
$this->assertTrue($dispatched);

But it reaches the same result so it’s just a matter of preference.

Reply:
I’m traveling and on my phone so this is real rough...

$dispatched = false;
App()->listen(MyEvent::class, function() {
  $dispatched = true;
});

$collector->startMeasure(...);
$this->assertTrue($dispatched);

But it reaches the same result so it’s just a matter of preference.

Reply:
I see. In that case, we are testing that not only the event was fired, but also if the data passed through the event is correct.

```php
$this->assertEquals(self::SPAN_NAME, $event->name);
$this->assertEquals('request', $event->type);
$this->assertNull($event->action);
$this->assertNull($event->label);
$this->assertNull($event->start_time);
```